openpype/modules/sync_server/launch_hooks/pre_copy_last_published_workfile.py

```diff
@@ -94,38 +96,52 @@ def execute(self):
 
         self.log.info("Trying to fetch last published workfile...")
 
-        project_doc = self.data.get("project_doc")
         asset_doc = self.data.get("asset_doc")
         anatomy = self.data.get("anatomy")
 
-        # Check it can proceed
-        if not project_doc and not asset_doc:
-            return
-
-        # Get subset id
-        subset_id = next(
-            (
-                subset["_id"]
-                for subset in get_subsets(
-                    project_name,
-                    asset_ids=[asset_doc["_id"]],
-                    fields=["_id", "data.family", "data.families"],
-                )
-                if subset["data"].get("family") == "workfile"
+        # Get subsets of the correct family
+        filtered_subsets = [
+            subset
+            for subset in get_subsets(
+                project_name,
+                asset_ids=[asset_doc["_id"]],
+                fields=["_id", "name", "data.family", "data.families"],
+            )
+            if (
+                subset["data"].get("family") == "workfile"
                 # Legacy compatibility
                 or "workfile" in subset["data"].get("families", {})
```

Comment:
Families should be a list right?
```suggestion
                or "workfile" in subset["data"].get("families", [])
```